tests/test_hdl_ast.py

```diff
@@ -1025,6 +1025,51 @@ def test_lower_to_user_value(self):
         self.assertEqual(uv.lower_count, 1)
 
 
+class MockValueCastableChanges(ValueCastable):
+    def __init__(self, width=0):
+        self.width = width
+
+    @ValueCastable.lowermethod
+    def as_value(self):
+        return Signal(self.width)
+
+
+class MockValueCastableNotDecorated(ValueCastable):
+    def __init__(self):
+        pass
+
+    def as_value(self):
+        return Signal()
+
+
+class MockValueCastableNoOverride(ValueCastable):
+    def __init__(self):
+        pass
+
+
+class ValueCastableTestCase(FHDLTestCase):
+    def test_not_decorated(self):
+        with self.assertRaisesRegex(TypeError,
+                r"^Classes deriving from `ValueCastable` must decorate the `as_value` "
+                "method with the `ValueCastable.lowermethod` decorator$"):
```

Comment:
Python string qualifies should be present for every continuation. I.e. if you concatenate two chunks of a raw string it looks like `r"foo" r"bar"`.